torch/jit/_script.py

```diff
@@ -1263,7 +1263,7 @@ def forward(self, input):
                 "Please directly inherit from 'object'."
             )
         if _rcb is None:
-            _rcb = _jit_internal.createResolutionCallbackFromFrame(_frames_up + 1)
+            _rcb = _jit_internal.createResolutionCallbackFromClosure(obj)
```

Comment:
I'm surprised that this actually works before, considering the difference between dynamic and lexical resolution. 

Could we rename them to `createResolutionCallbackFromLexicalScope` and `createResolutionCallbackFromDynamicScope` to avoid some misuse in the future?

Reply:
hmmmm not really, createResolutionCallbackFromFrame doesn't give you the lexical scope of the class; but the lexical scope at the point torch.jit.script is called.

Reply:
Update: after an online discussion, we think it's worthwhile to investigate why createResolutionCallbackFromFrame() was there in the first place. If it doesn't make sense we can just remove it, so that the new logic will be much simpler. If it's there for a reason, then I'm not keen on the renaming thing.

Reply:
So turns out createResolutionCallbackFromFrame cannot be removed because it is used to create compiled methods from raw source (as string) as well. In that case, there is no static scope because we dont know where the string come from.